src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -311,10 +311,6 @@ CitusBeginScanWithCoordinatorProcessing(CustomScanState *node, EState *estate, i
 	}
 	else if (jobQuery->commandType == CMD_SELECT && !workerJob->deferredPruning)
 	{
-		/* we'll use generated strings, no need to have the parameters anymore */
-		EState *executorState = planState->state;
-		ResetExecutionParameters(executorState);
-
 		/* we're done, we don't want to evaluate functions for SELECT queries */
```

Comment:
I feel like this code path is too convoluted and we should split it up by SELECT / modification, rather than processing / not processing.